manifests/pub_dir.pp

```diff
@@ -12,17 +12,38 @@
   String $pub_dir_options = '+FollowSymLinks +Indexes',
 ) {
   include foreman_proxy_content
-  include apache
 
   ensure_packages('katello-client-bootstrap')
 
-  apache::vhost { 'foreman_proxy_content':
-    servername          => $servername,
-    port                => 80,
-    priority            => '05',
-    docroot             => '/var/www/html',
-    options             => ['SymLinksIfOwnerMatch'],
-    additional_includes => ["${apache::confd_dir}/pulp-vhosts80/*.conf"],
-    custom_fragment     => template('foreman_proxy_content/httpd_pub.erb'),
+  if ($foreman_proxy_content::pulp_master or $foreman_proxy_content::pulp) {
+    $additional_includes = ["${apache::confd_dir}/pulp-vhosts80/*.conf"]
+  } else {
+    $additional_includes = []
+  }
+
+  if ($foreman_proxy_content::pulpcore or $foreman_proxy_content::pulp_master) {
+    include foreman::config::apache
+
+    foreman::config::apache::fragment { 'pub_dir':
+      content     => template('foreman_proxy_content/httpd_pub.erb'),
+      ssl_content => template('foreman_proxy_content/httpd_pub.erb'),
+    }
+  } elsif ($foreman_proxy_content::pulp) {
+    pulp::apache::fragment { 'pub_dir':
+      #content => template('foreman_proxy_content/httpd_pub.erb'),
```

Comment:
`pulp::apache::fragment` has no `content`, how do I get content into the 80 vhost?!

Reply:
wow, really, :shower: